spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, type: :controller do
+  let(:company) { create :company }
+  let(:user) { create :user, company: company }
+
+  before do
+    request.host = "#{company.subdomain}.example.com"
+    sign_in user
+
+    set_omniauth_intercom
+    get user_intercom_omniauth_authorize_path
```

Comment:
We should probably move these calls to the `it` block.
```suggestion
    set_omniauth_intercom
    get user_intercom_omniauth_authorize_path
```